src/server/req_register.c

```diff
@@ -294,64 +305,79 @@ req_register(struct batch_request *preq)
 			switch (type) {
 
 				case JOB_DEPEND_TYPE_AFTERSTART:
-					if (pjob->ji_qs.ji_substate >= JOB_SUBSTATE_RUNNING) {
-						/* job already running, setup task to send	*/
-						/* release back to child and continue with	*/
-						/* registration process 			*/
-						ptask = set_task(WORK_Immed, 0, post_run_depend,
-							(void *)pjob);
-						if (ptask)
-							append_link(&pjob->ji_svrtask,
-								&ptask->wt_linkobj, ptask);
-					}
+					if (is_finished == FALSE) {
+						if (pjob->ji_qs.ji_substate >= JOB_SUBSTATE_RUNNING) {
+							/* job already running, setup task to send	*/
+							/* release back to child and continue with	*/
+							/* registration process 			*/
+							ptask = set_task(WORK_Immed, 0, post_run_depend,
+								(void *)pjob);
+							if (ptask)
+								append_link(&pjob->ji_svrtask,
+									&ptask->wt_linkobj, ptask);
+						}
+					} else
+						rc = PBSE_HISTJOBID;
```

Comment:
Shouldn't you break after setting rc?

Reply:
You can also do the if (is_finished == TRUE){ rc = PBSE; break; } here

Reply:
Good catch :)